### DISCUSSION TITLE
Select All Button for Mantine - MultiSelect

### DISCUSSION THREAD

**Discussion by ks-himanshub** [mantinedev/mantine#3635] [Feature requests]
A select all button for multiselect component will be a cherry on the cake.

**Comment by rtivital** [comment]
You can implement this logic on your side

**Comment by enyelsequeira** [comment]
I was working on something similar make the multiselect have a select all option and with checkboxes, but haven’t been able to do it. Any luck?

**Reply by HimanshuBari21** [reply]
Not yet.

**Comment by omarlahkim** [comment]
You can do it using the state (hooks), add an ActionIcon or Button:

```
export default function Page(){
const [checkAll,setCheckAll] = useSate(false);
const [checked, setChecked] = react.useState([]);

function CheckAllBoxes() {
    if (checkAll) {
      setChecked([]);
      setCheckAll(false);
    } else {
      setChecked(orders.map((item) => item.id));
      setCheckAll(true);
    }
  }
```

`<ActionIcon size={"sm"} onClick={CheckAllBoxes}>
                <IconCircleCheckFilled />
              </ActionIcon>`

and the other checkboxes : 

`<Checkbox
                  checked={checked.includes(item.id)}
                  onChange={() => checkCheckbox(item.id)}
                />`
